### PR TITLE
Add versioning

### LINKED ISSUE #23: Proper version tagging
Hello.

Would it be possible to do a proper tagging for this project please? 

Maybe good starting point is here: https://github.com/actions/checkout 
They are using versions like:

* `v2` -> `v2.0.0`
* `v4` -> `v4.1.0`

It would be really handy to use something like `v4` instead of specifying the exact version only `v4.1.0`:

```yaml
    - uses: ansible/ansible-lint-action@v4
```

Thank you...

### PR DESCRIPTION
**Pull request by ruzickap**
Fix #23

### CODE REVIEW

File: .github/workflows/update-semver.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: update-semver
+
+on:
+  release:
+    types: [published]
+    tags:
+      - 'v*.*.*'
```
Comment (webknjaz) [inline .github/workflows/update-semver.yml:7]: Does `*` match any number of chars? Will it work with `.dev0`/`.post0` releases?

File: .github/workflows/update-semver.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  update-semver:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
```
Comment (webknjaz) [inline .github/workflows/update-semver.yml:13]: Why is this necessary?

File: .github/workflows/update-semver.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+on:
+  release:
+    types: [published]
+    tags:
+      - 'v*.*.*'
+
+jobs:
+  update-semver:
+    runs-on: ubuntu-18.04
```
Comment (webknjaz) [inline .github/workflows/update-semver.yml:11]: I prefer `latest` b/c GH tends to deprecate some envs, meaning we'd have to re-hardcode them.
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/update-semver.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Semver
+        uses: haya14busa/action-update-semver@v1
+        with:
+          major_version_tag_only: true
```
Comment (webknjaz) [inline .github/workflows/update-semver.yml:18]: Do you know how this'll handle pre-releases/dev/post?

File: .github/workflows/update-semver.yml (hunk 2 above)
Comment (ruzickap) [reply .github/workflows/update-semver.yml:13]: I took it from the official "README file":
https://github.com/haya14busa/action-update-semver

From what I can see it is running https://github.com/haya14busa/action-update-semver/blob/master/entrypoint.sh - yes it's necessary...

File: .github/workflows/update-semver.yml (hunk 1 above)
Comment (ruzickap) [reply .github/workflows/update-semver.yml:7]: Yes it should be the case: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

But I have never tried it...

Comment (ruzickap) [reply .github/workflows/update-semver.yml:7]: Sorry... I believe this will not match the regexp `v4.1.0.post0` - it means this will be ignored by GitHub action...

File: .github/workflows/update-semver.yml (hunk 4 above)
Comment (ruzickap) [reply .github/workflows/update-semver.yml:18]: It takes care only about the git tags.

I believe it will do tagging of `v4.1.0` -> `v4` and `v4.1.0.post0` will be "ignored".

Comment (webknjaz) [reply .github/workflows/update-semver.yml:18]: This is something that concerns me and I think this should be solved somehow before we can proceed.

Comment (ruzickap) [reply .github/workflows/update-semver.yml:18]: I believe only the "proper versions/releases" (like `1.2.3`) should be tagged with `@1`. 

According https://semver.org/ versions like `1.0.0-alpha`, `1.0.0-0.3.7` and other pre-leases should not be tagged by `@1`.

Anyway - choose the tools which suites you the best for the "releasing".

File: .github/workflows/update-semver.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+name: update-semver
```
Comment (kubin3818) [inline .github/workflows/update-semver.yml:1]: - name: ansible-lint
  uses: ansible/ansible-lint-action@v4.1.0.post0

Comment (ruzickap) [reply .github/workflows/update-semver.yml:1]: Do you want me to change the name to `ansible-lint` ?

But I do not understand what you mean by `uses: ansible/ansible-lint-action@v4.1.0.post0` ...

**Comment by ruzickap** [comment]
Any update on this topic or should I close this PR ?
